servlet/src/main/java/io/undertow/servlet/spec/ServletOutputStreamImpl.java

```diff
@@ -871,20 +889,13 @@ public void handleEvent(final StreamSinkChannel aChannel) {
                 try {
                     state |= FLAG_IN_CALLBACK;
 
-                    servletRequestContext.getCurrentServletContext().invokeOnWritePossible(servletRequestContext.getExchange(), listener);
-
-                    if (isReady()) {
-                        //if the stream is still ready then we do not resume writes
-                        //this is per spec, we only call the listener once for each time
-                        //isReady returns true
-                        if (channel != null) {
-                            channel.suspendWrites();
-                        }
-                    } else {
-                        if (channel != null) {
-                            channel.resumeWrites();
-                        }
+                    //if the stream is still ready then we do not resume writes
+                    //this is per spec, we only call the listener once for each time
+                    //isReady returns true
+                    if (channel != null) {
+                        channel.suspendWrites();
```

Comment:
Thanks for the explanation! I'd be happy to revisit this if we run into performance issues.